pkg/query-service/app/metricsexplorer/summary.go

```diff
@@ -66,3 +71,128 @@ func (receiver *SummaryService) FilterValues(ctx context.Context, params *metric
 		return nil, nil
 	}
 }
+
+func (receiver *SummaryService) GetMetricsSummary(ctx context.Context, metricName string) (metrics_explorer.MetricDetailsDTO, *model.ApiError) {
+	var (
+		wg               sync.WaitGroup
+		metricDetailsDTO metrics_explorer.MetricDetailsDTO
+		errCh            = make(chan *model.ApiError, 1)
+	)
+
+	// Create a context with cancellation to stop other goroutines on the first error
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	handleError := func(err *model.ApiError) {
+		select {
+		case errCh <- err: // Send the error if no error was sent yet
+			cancel() // Cancel other goroutines
+		default:
+		}
+	}
+
+	wg.Add(6)
```

Comment:
Would it be better to add `wg.Add(1)` inside the routine than the fixed number here?